password-hash/src/errors.rs

```diff
@@ -56,10 +56,23 @@ pub enum Error {
     /// Salt too long.
     SaltTooLong,
 
+    /// Salt invalid.
+    SaltInvalid,
+
     /// Invalid algorithm version.
     Version,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum ParamValueError {
```

Comment:
What about calling this `InvalidValue` instead, and also parameterizing `Error::SaltInvalid` with it, then eliminating `SaltTooShort` and `SaltTooLong`?

I say this because at least in the current implementation salts and parameter values follow the same rules, so it would be nice to deduplicate the error handling for them.

Reply:
Yeah, that makes sense and allows to simplify things a little bit. I have adjusted it.